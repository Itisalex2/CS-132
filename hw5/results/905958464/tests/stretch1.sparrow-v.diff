0a1,4
> 100
> 40464
> 80770
> 121302
