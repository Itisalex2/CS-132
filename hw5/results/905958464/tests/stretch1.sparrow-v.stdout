.equiv @sbrk, 9
.equiv @print_string, 4
.equiv @print_char, 11
.equiv @print_int, 1
.equiv @exit, 10
.equiv @exit2, 17

.text

.globl main
    jal Main
    li a0, @exit
    ecall

.globl Main
Main:
    sw   fp, -8(sp)
    mv   fp, sp
    addi sp, sp, -16
    sw   ra, -4(fp)
    la   t0, f
    sw   t0, -12(fp)
    lw   t1, -12(fp)
    jalr t1
    mv   t0, a0
    sw   t0, -16(fp)
    lw   t1, -16(fp)
    mv   a0, t1
    jal  print
    lw   a0, -16(fp)
    lw   ra, -4(fp)
    lw   fp, -8(fp)
    addi sp, sp, 16
    jr   ra

.globl f
f:
    sw   fp, -8(sp)
    mv   fp, sp
    addi sp, sp, -156
    sw   ra, -4(fp)
    li   t0, 10
    sw   t0, -12(fp)
    li   t0, 20
    sw   t0, -16(fp)
    lw   t1, -12(fp)
    lw   t2, -16(fp)
    add  t0, t1, t2
    sw   t0, -20(fp)
    li   t0, 30
    sw   t0, -24(fp)
    lw   t1, -12(fp)
    lw   t2, -24(fp)
    slt  t0, t1, t2
    sw   t0, -28(fp)
    lw   t1, -28(fp)
    bnez t1, f_no_long_jump0
    jal  f_L8
f_no_long_jump0:
    lw   t1, -12(fp)
    lw   t2, -20(fp)
    add  t0, t1, t2
    sw   t0, -32(fp)
    lw   t1, -12(fp)
    lw   t2, -16(fp)
    add  t0, t1, t2
    sw   t0, -36(fp)
    jal  f_L10
f_L8:
    lw   t1, -16(fp)
    lw   t2, -20(fp)
    add  t0, t1, t2
    sw   t0, -40(fp)
    jal  f_L14
f_L10:
    li   t0, 50
    sw   t0, -44(fp)
    lw   t1, -32(fp)
    lw   t2, -44(fp)
    add  t0, t1, t2
    sw   t0, -16(fp)
    lw   t1, -20(fp)
    lw   t2, -32(fp)
    add  t0, t1, t2
    sw   t0, -48(fp)
    lw   t1, -48(fp)
    lw   t2, -36(fp)
    add  t0, t1, t2
    sw   t0, -52(fp)
    lw   t1, -52(fp)
    mv   a0, t1
    jal  print
    lw   t1, -20(fp)
    lw   t2, -36(fp)
    add  t0, t1, t2
    sw   t0, -40(fp)
    li   t0, 40
    sw   t0, -56(fp)
    lw   t1, -16(fp)
    lw   t2, -56(fp)
    add  t0, t1, t2
    sw   t0, -16(fp)
    jal  f_L8
f_L14:
    li   t0, 10
    sw   t0, -60(fp)
    lw   t1, -12(fp)
    lw   t2, -60(fp)
    add  t0, t1, t2
    sw   t0, -36(fp)
f_L22:
    li   t0, 55
    sw   t0, -16(fp)
    li   t0, 20
    sw   t0, -64(fp)
    lw   t1, -12(fp)
    lw   t2, -64(fp)
    sub  t0, t1, t2
    sw   t0, -20(fp)
    lw   t1, -16(fp)
    lw   t2, -20(fp)
    slt  t0, t1, t2
    sw   t0, -68(fp)
    lw   t1, -68(fp)
    bnez t1, f_no_long_jump1
    jal  f_L20
f_no_long_jump1:
    lw   t1, -20(fp)
    lw   t2, -16(fp)
    add  t0, t1, t2
    sw   t0, -32(fp)
    lw   t1, -32(fp)
    lw   t2, -60(fp)
    sub  t0, t1, t2
    sw   t0, -40(fp)
    lw   t1, -32(fp)
    lw   t2, -40(fp)
    add  t0, t1, t2
    sw   t0, -72(fp)
    lw   t1, -72(fp)
    mv   a0, t1
    jal  print
    lw   t1, -20(fp)
    lw   t2, -40(fp)
    add  t0, t1, t2
    sw   t0, -16(fp)
    li   t0, 2
    sw   t0, -76(fp)
    lw   t1, -76(fp)
    lw   t2, -16(fp)
    sub  t0, t1, t2
    sw   t0, -36(fp)
    jal  f_L22
f_L20:
    li   t0, 7
    sw   t0, -80(fp)
    lw   t1, -36(fp)
    lw   t2, -80(fp)
    add  t0, t1, t2
    sw   t0, -16(fp)
    lw   t1, -20(fp)
    lw   t2, -16(fp)
    add  t0, t1, t2
    sw   t0, -20(fp)
    li   t0, 2
    sw   t0, -76(fp)
    lw   t1, -76(fp)
    lw   t2, -20(fp)
    mul  t0, t1, t2
    sw   t0, -40(fp)
    lw   t1, -36(fp)
    lw   t2, -60(fp)
    add  t0, t1, t2
    sw   t0, -84(fp)
    li   t0, 55
    sw   t0, -88(fp)
    lw   t1, -88(fp)
    sw   t1, -92(fp)
    la   t0, g
    sw   t0, -96(fp)
    lw   t1, -84(fp)
    lw   t2, -92(fp)
    slt  t0, t1, t2
    sw   t0, -24(fp)
    lw   t1, -24(fp)
    bnez t1, f_no_long_jump2
    jal  f_L27
f_no_long_jump2:
    lw   t1, -84(fp)
    lw   t2, -64(fp)
    sub  t0, t1, t2
    sw   t0, -100(fp)
    lw   t1, -96(fp)
    addi sp, sp, -32
    lw   t6, -16(fp)
    sw   t6, 0(sp)
    lw   t6, -20(fp)
    sw   t6, 4(sp)
    lw   t6, -32(fp)
    sw   t6, 8(sp)
    lw   t6, -40(fp)
    sw   t6, 12(sp)
    lw   t6, -36(fp)
    sw   t6, 16(sp)
    lw   t6, -84(fp)
    sw   t6, 20(sp)
    lw   t6, -92(fp)
    sw   t6, 24(sp)
    lw   t6, -100(fp)
    sw   t6, 28(sp)
    jalr t1
    addi sp, sp, 32
    mv   t0, a0
    sw   t0, -92(fp)
    lw   t1, -100(fp)
    lw   t2, -92(fp)
    add  t0, t1, t2
    sw   t0, -104(fp)
    jal  f_L29
f_L27:
    li   t0, 1
    sw   t0, -108(fp)
    lw   t1, -84(fp)
    lw   t2, -108(fp)
    add  t0, t1, t2
    sw   t0, -112(fp)
    lw   t1, -76(fp)
    lw   t2, -112(fp)
    mul  t0, t1, t2
    sw   t0, -104(fp)
f_L29:
    li   t0, 5
    sw   t0, -116(fp)
    lw   t1, -104(fp)
    lw   t2, -116(fp)
    sub  t0, t1, t2
    sw   t0, -120(fp)
    lw   t1, -84(fp)
    lw   t2, -120(fp)
    add  t0, t1, t2
    sw   t0, -112(fp)
    lw   t1, -104(fp)
    lw   t2, -112(fp)
    sub  t0, t1, t2
    sw   t0, -124(fp)
    lw   t1, -124(fp)
    lw   t2, -120(fp)
    add  t0, t1, t2
    sw   t0, -128(fp)
    lw   t1, -128(fp)
    mv   a0, t1
    jal  print
    lw   t1, -104(fp)
    lw   t2, -112(fp)
    add  t0, t1, t2
    sw   t0, -100(fp)
    lw   t1, -20(fp)
    lw   t2, -32(fp)
    add  t0, t1, t2
    sw   t0, -20(fp)
    lw   t1, -32(fp)
    lw   t2, -64(fp)
    add  t0, t1, t2
    sw   t0, -132(fp)
    lw   t1, -100(fp)
    lw   t2, -132(fp)
    add  t0, t1, t2
    sw   t0, -136(fp)
    lw   t1, -32(fp)
    lw   t2, -136(fp)
    slt  t0, t1, t2
    sw   t0, -48(fp)
f_L3:
    lw   t1, -48(fp)
    bnez t1, f_no_long_jump3
    jal  f_L12
f_no_long_jump3:
    lw   t1, -32(fp)
    lw   t2, -132(fp)
    add  t0, t1, t2
    sw   t0, -140(fp)
    li   t0, 30
    sw   t0, -144(fp)
    lw   t1, -140(fp)
    lw   t2, -144(fp)
    add  t0, t1, t2
    sw   t0, -148(fp)
    lw   t1, -100(fp)
    lw   t2, -148(fp)
    sub  t0, t1, t2
    sw   t0, -148(fp)
    lw   t1, -140(fp)
    lw   t2, -148(fp)
    sub  t0, t1, t2
    sw   t0, -136(fp)
    li   t0, 0
    sw   t0, -152(fp)
    lw   t1, -136(fp)
    lw   t2, -152(fp)
    mul  t0, t1, t2
    sw   t0, -48(fp)
    lw   t1, -20(fp)
    lw   t2, -84(fp)
    add  t0, t1, t2
    sw   t0, -132(fp)
    lw   t1, -140(fp)
    lw   t2, -136(fp)
    add  t0, t1, t2
    sw   t0, -136(fp)
    lw   t1, -132(fp)
    lw   t2, -136(fp)
    sub  t0, t1, t2
    sw   t0, -156(fp)
    lw   t1, -156(fp)
    mv   a0, t1
    jal  print
    jal  f_L3
f_L12:
    lw   t1, -112(fp)
    lw   t2, -136(fp)
    sub  t0, t1, t2
    sw   t0, -84(fp)
    lw   t1, -140(fp)
    lw   t2, -84(fp)
    add  t0, t1, t2
    sw   t0, -84(fp)
    lw   a0, -84(fp)
    lw   ra, -4(fp)
    lw   fp, -8(fp)
    addi sp, sp, 156
    jr   ra

.globl g
g:
    sw   fp, -8(sp)
    mv   fp, sp
    addi sp, sp, -60
    sw   ra, -4(fp)
    lw   t1, 0(fp)
    lw   t2, 4(fp)
    add  t0, t1, t2
    sw   t0, -12(fp)
    lw   t1, 8(fp)
    lw   t2, 12(fp)
    add  t0, t1, t2
    sw   t0, -16(fp)
    lw   t1, 16(fp)
    lw   t2, 20(fp)
    add  t0, t1, t2
    sw   t0, -20(fp)
    lw   t1, 24(fp)
    lw   t2, 28(fp)
    add  t0, t1, t2
    sw   t0, -24(fp)
    li   t0, 10
    sw   t0, -28(fp)
    lw   t1, -28(fp)
    sw   t1, -32(fp)
    lw   t1, 28(fp)
    lw   t2, 0(fp)
    slt  t0, t1, t2
    sw   t0, -36(fp)
    lw   t1, -36(fp)
    bnez t1, g_no_long_jump4
    jal  g_L42
g_no_long_jump4:
    la   t0, g
    sw   t0, -40(fp)
    lw   t1, -40(fp)
    addi sp, sp, -32
    lw   t6, -36(fp)
    sw   t6, 0(sp)
    lw   t6, 24(fp)
    sw   t6, 4(sp)
    lw   t6, -36(fp)
    sw   t6, 8(sp)
    lw   t6, 16(fp)
    sw   t6, 12(sp)
    lw   t6, -28(fp)
    sw   t6, 16(sp)
    lw   t6, -36(fp)
    sw   t6, 20(sp)
    lw   t6, 4(fp)
    sw   t6, 24(sp)
    lw   t6, -28(fp)
    sw   t6, 28(sp)
    jalr t1
    addi sp, sp, 32
    mv   t0, a0
    sw   t0, -12(fp)
g_L42:
    lw   t1, -24(fp)
    lw   t2, -12(fp)
    sub  t0, t1, t2
    sw   t0, -32(fp)
    lw   t1, -16(fp)
    lw   t2, -32(fp)
    mul  t0, t1, t2
    sw   t0, -44(fp)
    lw   t1, -20(fp)
    lw   t2, -44(fp)
    sub  t0, t1, t2
    sw   t0, -48(fp)
    lw   t1, -24(fp)
    lw   t2, -32(fp)
    add  t0, t1, t2
    sw   t0, -52(fp)
    lw   t1, -32(fp)
    lw   t2, -44(fp)
    add  t0, t1, t2
    sw   t0, -56(fp)
    lw   t1, -48(fp)
    lw   t2, -52(fp)
    add  t0, t1, t2
    sw   t0, -60(fp)
    lw   a0, -60(fp)
    lw   ra, -4(fp)
    lw   fp, -8(fp)
    addi sp, sp, 60
    jr   ra

.globl print
print:
    mv a1, a0
    li a0, @print_int
    ecall
    li a1, 10
    li a0, @print_char
    ecall
    jr ra

.globl error
error:
    mv a1, a0
    li a0, @print_string
    ecall
    li a1, 10
    li a0, @print_char
    ecall
    li a0, @exit
    ecall
abort_17:
    j abort_17

.globl alloc
alloc:
    mv a1, a0
    li a0, @sbrk
    ecall
    jr ra

.data

.globl msg_nullptr
msg_nullptr:
    .asciiz "null pointer"
    .align 2

.globl msg_array_oob
msg_array_oob:
    .asciiz "array index out of bounds"
    .align 2

